test/framework/README.md

```diff
@@ -0,0 +1,38 @@
+# CAPI e2e testing framework
+
+This framework aims to define common end-to-end patterns that can be reused across Cluster API providers.
+
+## Usage
+
+Use this framework as you see fit. If there are pieces that don't work for you, do not use them. If there are pieces
+that almost work for you, customize them. If this does not fit your e2e testing please file an issue and we can discuss
+your use case and find a nice solution for everyone.
+
+### Contents
+
+This framework contains
+
+* An interface to a management cluster
+    * An struct that implements the management cluster interface using `kind` as the backend.
+* A series of behavioral tests
+* Clean up tests.
+
+## Requirements
+
+### Code
+
+* Your provider CRDs must implement [`runtime.Object`][runtime.object]. This should already be true if you are using
```

Comment:
Is this even worth documenting? Can a CRD not be a `runtime.Object`?

Reply:
Interesting, I could certainly remove it. Is this harmful to keep?

Reply:
I think so? Could be misleading to folks who haven't had much exposure to apimachinery

Reply:
I would remove this. Assuming you're defining your types using Kubernetes apimachinery, all API types are automatically `runtime.Object`.

Reply:
ok, two for removal, 👋 words